src/app/modules/candidate-test-link/candidate-test-link.component.html

```diff
@@ -0,0 +1,8 @@
+    <div class="main">
+      <div class="logo">
+        <img src="assets/logo-high-res.png">
+      </div>
+      <button mat-button class="test-link" (click)="generateTestLink('app-deep-link')">
```

Comment:
app-deep-link is static so just use it as it is in component file 
e.g.
(click)="generateTestLink()"
ts file
generateTestLink() {
 url-app-deep-link
}